src/bin/stratis_dumpmetadata.rs

```diff
@@ -0,0 +1,37 @@
+use std::{
+    env,
+    fs::OpenOptions,
+    io::{Seek, SeekFrom},
+    process,
+    vec::Vec,
+};
+
+use serde_json::Value;
+
+use libstratis::engine::BDA;
+
+fn run(devpath: String) {
+    let mut devfile = OpenOptions::new().read(true).open(&devpath).unwrap();
+
+    let bda = BDA::load(&mut devfile).unwrap().unwrap();
+    println!("{:#?}", bda);
+
+    devfile.seek(SeekFrom::Start(0)).unwrap();
+
+    let loaded_state = bda.load_state(&mut devfile).unwrap();
+    println!("State JSON data:");
+    let state_json: Value = serde_json::from_slice(&loaded_state.unwrap()).unwrap();
+    let state_json_pretty: String = serde_json::to_string_pretty(&state_json).unwrap();
+    println!("{}", state_json_pretty);
+}
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+    if args.len() < 2 {
+        println!("Usage: stratis_dumpmetadata <device>");
+        process::exit(1);
+    }
+    let devpath = args[1].clone();
+
+    run(devpath);
```

Comment:
This line here will have a compiler error associated with it. Because ```run(devpath)``` now evaluates to a ```Result``` you'll need to match the ```Result``` and do the right thing depending on whether it is ```Err``` or ```Ok```.